src/config/sled.rs

```diff
@@ -460,6 +464,112 @@ impl IdentityKeyStore for SledConfigStore {
     }
 }
 
+fn thread_key(t: &Thread) -> Vec<u8> {
+    let mut bytes = SLED_TREE_THREAD_PREFIX.as_bytes().to_owned();
+    bytes.append(&mut t.into());
+    bytes
+}
+
+impl MessageStore for SledConfigStore {
+    type MessagesIter = SledMessagesIter;
+    fn save_message(
+        &mut self,
+        message: libsignal_service::prelude::Content,
+        receiver: Option<impl Into<ServiceAddress>>,
+    ) -> Result<(), Error> {
+        let receiver = receiver.map(|s| s.into());
+
+        let id = MessageIdentity::try_from(&message)?;
+        let thread = Thread::from_content_receiver(&message, receiver.as_ref())?;
+        let timestamp = id.1;
+
+        log::trace!("Storing a message with id: {:?}", id);
+
+        let tree_messages = self
+            .db
+            .try_read()
+            .expect("poisoned mutex")
+            .open_tree(SLED_TREE_MESSAGES)?;
+        let tree_thread = self
+            .db
+            .try_read()
+            .expect("poisoned mutex")
+            .open_tree(thread_key(&thread))?;
+
+        let key: [u8; 24] = id.into();
+        let value = ContentProto::from_content(message);
+        let value_vec = value.encode_to_vec();
+
+        tree_messages.insert(key, value_vec.clone())?;
+        tree_thread.insert(timestamp.to_be_bytes(), value_vec)?;
+        Ok(())
+    }
+
+    fn messages(&self) -> Result<Vec<libsignal_service::prelude::Content>, Error> {
+        Ok(self
+            .db
+            .read()
+            .expect("poisoned mutex")
+            .open_tree(SLED_TREE_MESSAGES)?
+            .iter()
+            .filter_map(Result::ok)
+            .filter_map(|(_key, buf)| ContentProto::decode(&*buf).ok())
+            .map(|c| c.into_content())
+            .collect())
+    }
+
+    fn message_by_identity(
+        &self,
+        id: &MessageIdentity,
+    ) -> Result<Option<libsignal_service::prelude::Content>, Error> {
+        let tree = self
+            .db
+            .try_read()
+            .expect("poisoned mutex")
+            .open_tree(SLED_TREE_MESSAGES)?;
+        // Big-Endian needed, otherwise wrong ordering in sled.
+        let key: [u8; 24] = id.clone().into();
+        let val = tree.get(key)?;
+        if let Some(val) = val {
+            let proto = ContentProto::decode(&*val)?;
+            Ok(Some(proto.into_content()))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn messages_by_thread(
```

Comment:
Following the previous comment, I would rename this method `messages` with the same parameters.